### PR TITLE
[Spaceship] Get and set territories

### PR DESCRIPTION
**Pull request by andyvanosdale**
Allow for iOS territory updating with Spaceship [https://github.com/fastlane/fastlane/issues/6592]

### CODE REVIEW

**Comment by enozero** [comment]
Hello! Any thoughts or questions about this request?

**Comment by ohwutup** [comment]
Hey @enozero! Apologies for the delay here, would you mind rebasing on master and we can take a closer look at this PR without merge conflicts? Thanks!! 👍 🐙

**Comment by enozero** [comment]
@ohwutup done

**Comment by kevinlondon** [comment]
This will be great because it will enable us to programmatically remove apps from sale 👍

File: spaceship/lib/spaceship/tunes/application.rb [hunk 1]
```diff
@@ -239,6 +239,16 @@ def price_tier
         client.price_tier(self.apple_id)
       end
 
+      # set the availability. This method doesn't require `save` to be called
+      def update_availability!(territories, opts = {})
```
Comment (mpirri) [inline spaceship/lib/spaceship/tunes/application.rb]: In this implementation, `update_availability` does not take the same type of object as what is returned by `availability`. I.e., `update_availability` takes an array of territories, while `availability` returns an `Availability` object.

So, you can't do something like this without an error:
```
availability = app.availability
app.update_availability!(availability)
```
I could see this as a source of confusion for users.

Some ideas:
- Make `update_availability` take an `Availability` object and create a constructor for `Availability` that takes an array of territories and an optional `include_future_territories` value.
- Split the `availability` accessors into separate `availability_territories` and `availability_include_future_territories` accessors

Thoughts?

**Review by mpirri — CHANGES_REQUESTED** [review]
This is awesome functionality, I can't wait to get this merged.  So sorry about the delay with reviewing this!

File: spaceship/lib/spaceship/tunes/application.rb (hunk 1 above)
Comment (enozero) [inline spaceship/lib/spaceship/tunes/application.rb]: Thanks for the review! I completely see where you are coming from and like that approach.

**Comment by enozero** [comment]
@mpirri I have updated the PR with your requested changes. Please let me know if you have any questions about them!

**Comment by mpirri** [comment]
@enozero Are you sure you committed/pushed up changes to `application.rb`?  It looks to me like it is out of sync with `tunes_client.rb` - for instance, the number of arguments it is passing to `client.update_availability!()` is inconsistent - 3 passed in `application.rb` and 2 accepted in `tunes_client.rb`.

**Comment by enozero** [comment]
@mpirri Great catch! I forgot to update the `Application` class. That has been updated (along with new added tests). The commit has been updated. Sorry about that!

**Review by mpirri — APPROVED** [review]

**Comment by mpirri** [comment]
This looks awesome!  Thanks for your work (and patience) on this, @enozero .  🚀

**Comment by enozero** [comment]
Thanks for the review! Looking forward to contributing more to the project.

**Pull request merged**